client/basic-authorship/src/basic_authorship.rs

```diff
@@ -459,15 +501,26 @@ mod tests {
 			block
 		};
 
+		futures::executor::block_on(
+			txpool.maintain(chain_event(
+				0,
+				client.header(&BlockId::Number(0u64)).expect("header get error").expect("there should be header")
+			))
+		);
+
 		// let's create one block and import it
 		let block = propose_block(&client, 0, 2, 7);
 		client.import(BlockOrigin::Own, block).unwrap();
 
-		// now let's make sure that we can still make some progress
+		futures::executor::block_on(
+			txpool.maintain(chain_event(
+				1,
+				client.header(&BlockId::Number(1)).expect("header get error").expect("there should be header")
+			))
+		);
 
-		// This is most likely incorrect, and caused by #5139
-		let tx_remaining = 0;
-		let block = propose_block(&client, 1, 2, tx_remaining);
+		// now let's make sure that we can still make some progress
+		let block = propose_block(&client, 1, 2, 5);
```

Comment:
Why did we had before `0` remaining and now `5`?

Reply:
There were 7 initially, and 2 was included in previous block
(see deleted comment above that outlines that 0 was an incorrect observation caused by the bug this pr aims to fix among other things)